easybuild/easyconfigs/c/Charm++/Charm++-gerrit1886-CrayGNU-2016.11.eb

```diff
@@ -0,0 +1,31 @@
+easyblock = 'CmdCp'
+
+name = 'Charm++'
+version = 'gerrit1886'
+
+homepage = 'http://charm.cs.illinois.edu/research/charm/'
+description = """Charm++ is a machine independent parallel programming system. Programs written using this system will run unchanged on MIMD machines with or without a shared memory."""
+
+toolchain = {'name': 'CrayGNU', 'version': '2016.11'}
+toolchainopts = {'dynamic': False}
+
+# source_urls = ['http://charm.cs.illinois.edu/distrib/']
+sources = ['charm-%(version)s.tar.gz']
+cmds_map = [('charm-%(version)s.tar.gz', './build ChaNGa gni-crayxc cuda hugepages --enable-lbuserdata --with-production -j12')]
+patches = [ 'Charm++-gerrit1886-CrayGNU-2016.11.patch' ]
```

Comment:
Did you try to add the flags "$CRAY_UGNI_POST_LINK_OPTS -lugni $CRAY_PMI_POST_LINK_OPTS -lpmi" ?

With that you might be able to skip this patch.